go-runtime/ftl/config.go

```diff
@@ -15,26 +15,25 @@ type ConfigType interface{ any }
 // Config declares a typed configuration key for the current module.
 func Config[T ConfigType](name string) ConfigValue[T] {
 	module := callerModule()
-	return ConfigValue[T]{module, name}
+	return ConfigValue[T]{Ref{module, name}}
 }
 
 // ConfigValue is a typed configuration key for the current module.
 type ConfigValue[T ConfigType] struct {
-	module string
-	name   string
+	Ref
```

Comment:
Let's make this private: `ref Ref`

Reply:
This leads to the issue of how to read the name in ftltest to make this one work:
https://github.com/TBD54566975/ftl/blob/1313800fed2ff93aeddf02a0925f1227f4f5ded4/go-runtime/ftl/ftltest/ftltest.go#L52
exposing a Ref() or Name() func goes against the idea of making this private right?

Reply:
Makes sense